### DISCUSSION TITLE
[TypeError: Cannot read properties of undefined (reading 'text')]

### DISCUSSION THREAD

**Discussion by ghost** [vercel/next.js#54362] [Help]
### Summary

I use Cloudflare Pages + Workers to create Edge applications.
When I enable the experimental-edge runtime to use Dynamic Routes, I get the following error on the console and the website just says "Internal Server Error."

```ts
[TypeError: Cannot read properties of undefined (reading 'text')]
```

### Additional information

```js
Operating System:
      Platform: win32
      Arch: x64
      Version: Windows 10 Pro
    Binaries:
      Node: 18.12.1
      npm: N/A
      Yarn: N/A
      pnpm: N/A
    Relevant Packages:
      next: 13.4.13
      eslint-config-next: 13.4.13
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 5.1.6
    Next.js Config:
      output: N/A
```

**Comment by ghost** [comment]
As a trial, I deployed it to Cloudflare and it worked.
For some reason GetStaticProps and GetServerSideProps do not work only in my environment.
Removing them will work, but not if they are included in the code, resulting in an Internal Server Error.

**Comment by icyJoseph** [comment]
Could you try to produce a minimal page example that crashes and post it back here? I tried but it works fine locally. You did say, experimental-edge, but I think we have to use `edge` in the latest versions:

- https://nextjs.org/docs/pages/building-your-application/data-fetching/get-server-side-props#getserversideprops-with-edge-api-routes

**Reply by ghost** [reply]
I will create and upload it now.

I'm using experimental-edge because the console says "-error Page/[slug] provided runtime 'edge', the edge runtime for rendering is currently experimental.Use runtime 'experimental-edge' instead." when I use edge.

**Reply by ghost** [reply]
https://github.com/KohnoseLami/my-app/blob/main/src/pages/%5Bslug%5D.tsx

Done
Combining edge runtime with dynamic routes results in an error.

**Comment by ghost** [comment]
https://github.com/KohnoseLami/my-app/blob/main/src/pages/%5Bslug%5D.tsx

Created application with minimal configuration
That's when the error changed.
```
[Error: <Html> should not be imported outside of pages/_document.
Read more: https://nextjs.org/docs/messages/no-document-import-in-page]
```

Unable to access local files when using Edge Runtime?
